### PR TITLE
Typical Use-Cases section [7899]

### PR DESCRIPTION
**Pull request by EduPonz**
This PR add a **Typical Use-Cases** section to the documentation containing the following:

* Fast-RTPS over WIFI
** Initial peers
** Disabling Multicast Discovery
* Wide Deployments
** Server-Client Discovery
** Well-known Network Topologies
* Fast-RTPS in ROS 2
** Configuring Fast-RTPS in ROS 2 with XML files

### CODE REVIEW

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/183/)

File: docs/use-cases.rst [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+Wide Deployments
+================
+
+Systems with large amounts of communication nodes might pose a challenge to
+`Data Distribution Service (DDS) <https://www.omg.org/spec/DDS/1.4/PDF>`_ based middleware implementations in terms of
+setup times, memory consumption, and network load. This is because, as explained in :ref:`discovery`, the Participant
+Discovery Phase (PDP) relies on meta traffic announcements send to multicast addresses so that all the participants in
+the network can acknowledge each other. This phase is followed by a Endpoint Discovery Phase (EDP) where all the
+participants exchange information (using unicast addresses) about their publisher and subscriber entities with the rest
+of the participants, so that matching between publishers and subscribers in the same topic can occur. As the number of
```
[the remark this answers is not in the document]
Comment (EduPonz) [reply docs/use-cases.rst]: I think it's clearer the way it is

File: docs/use-cases.rst [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+-----------------------------
+
+If all the peers are known beforehand, it is possible to disable the multicast meta traffic completely. This is done
+using the configuration attribute ``metatrafficUnicastLocatorList``. By defining a custom
+``metatrafficUnicastLocatorList``, the default metatraffic multicast and unicast locators to be employed by the
+participant are avoided, which prevents the participant from listening to any discovery data from multicast sources.
+The local participant creates a meta traffic receiving resource per address-port pair specified in the
+``metatrafficUnicastLocatorList``.
+
+Consideration should be given to the assignment of the address-port pair in the ``metatrafficUnicastLocatorList``,
```
[the remark this answers is not in the document]
Comment (EduPonz) [reply docs/use-cases.rst]: Why the suggestion?

File: docs/use-cases.rst [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+.. _use-case-initial-peers:
+
+Initial Peers
+-------------
+
+According to the `RTPS standard <https://www.omg.org/spec/DDSI-RTPS/2.2/PDF>`_ (Section 9.6.1.1), each participant must
+listen for incoming
+PDP discovery metatraffic in two different ports, one linked with a multicast address, and another one linked to a
+unicast address.
+Fast-RTPS allows for the configuration of an initial peers list which contains one or much such address-port pairs
```
[the remark this answers is not in the document]
Comment (EduPonz) [reply docs/use-cases.rst]: Suggestions?

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/199/)

**Pull request merged**